WeTransfer/format_parser
# Facilitate files analysis for ActiveStorage

Solution for enhancement https://github.com/WeTransfer/format_parser/issues/93

**Assumptions:**

- for `ActiveStorage::Analyzer::ImageAnalyzer`:
  - [rescue block](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/analyzer/image_analyzer.rb#L28) handling **lack of mini_magick gem** has been removed since it's not needed anymore.
  - [private method](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/analyzer/image_analyzer.rb#L41) `#rotated_image?(image)` has been removed:
    - Swapping image height & width if image is rotated is already handled in parsers of `format_parser` gem, for example:
    https://github.com/WeTransfer/format_parser/blob/master/lib/parsers/tiff_parser.rb#L28

**Note:**

Test specs are not there yet because:

- For `require_relative 'railtie' if defined?(Rails)`: If we remove `if defined?(Rails)` part, all existing test specs will fail with the following error:
```
uninitialized constant FormatParser::ActiveStorage::Rails
```

Which makes sense; because we are running the specs, while we are not having a Rails app.

- If we want to add specs for the Railtie, I can see these options:
  - Build a small Rails app inside the spec directory and use it for testing the Railtie (I am in favor of this option since we have only this case ATM).
  - Use this gem https://github.com/pat/combustion, which is achieving the same purpose without the cost of creating small Rails app.
  - Maybe there is no need to test it out, since now we use methods from `format_parser` that are already test-covered.
  - Another option?


@julik  @dennisvdvliet  Please let me know your thoughts.
